cmd/kaspawallet/daemon/server/sync.go

```diff
@@ -0,0 +1,257 @@
+package server
+
+import (
+	"github.com/kaspanet/kaspad/app/appmessage"
+	"github.com/kaspanet/kaspad/domain/consensus/model/externalapi"
+	"github.com/pkg/errors"
+	"time"
+)
+
+const (
+	// externalKeychain is the key chain that is used to create receive addresses
+	externalKeychain = 0
+	// internalKeychain is used to create change addresses
+	internalKeychain = 1
+)
+
+var keyChains = []uint8{externalKeychain, internalKeychain}
+
+type walletAddressSet map[string]*walletAddress
+
+func (was walletAddressSet) strings() []string {
+	addresses := make([]string, 0, len(was))
+	for addr := range was {
+		addresses = append(addresses, addr)
+	}
+	return addresses
+}
+
+func (s *server) sync() error {
+	ticker := time.NewTicker(time.Second)
+	defer ticker.Stop()
+
+	for range ticker.C {
+		err := s.collectUTXOsFromRecentAddresses()
+		if err != nil {
+			return err
+		}
+
+		err = s.collectUTXOsFromFarAddresses()
+		if err != nil {
+			return err
+		}
+
+		err = s.refreshExistingUTXOsWithLock()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+const numIndexesToQuery = 100
+
+// addressesToQuery scans the addresses in the given range. Because
+// each cosigner in a multisig has its own unique path for generating
+// addresses it goes over all the cosigners and add their addresses
+// for each key chain.
+func (s *server) addressesToQuery(start, end uint32) (walletAddressSet, error) {
+	addresses := make(walletAddressSet)
+	for index := start; index < end; index++ {
+		for cosignerIndex := uint32(0); cosignerIndex < uint32(len(s.keysFile.ExtendedPublicKeys)); cosignerIndex++ {
+			for _, keychain := range keyChains {
+				address := &walletAddress{
+					index:         index,
+					cosignerIndex: cosignerIndex,
+					keyChain:      keychain,
+				}
+				addressString, err := s.walletAddressString(address)
+				if err != nil {
+					return nil, err
+				}
+				addresses[addressString] = address
+			}
+		}
+	}
+
+	return addresses, nil
+}
+
+// collectUTXOsFromFarAddresses collects UTXOs
+// from s.nextSyncStartIndex to s.nextSyncStartIndex+numIndexesToQuery
```

Comment:
I'd rather if these comments were more accessible (a.k.a. didn't assume person knows the meaning of variables such as s.nextSyncStartIndex.

In other words, explain the difference between Close and Far addresses in laymen terms.